### PR TITLE
169395632 v7 2 changes

### PR DESCRIPTION
**Pull request by davidbl**
PT story https://www.pivotaltracker.com/story/show/169395632
Associated doximity PR https://github.com/doximity/doximity/pull/30797

Overview
--------
There are breaking changes in Elasticsearch v7.0 that we need to address in order to upgrade

Technical Details
-----------------
Additional information available here https://docs.google.com/document/d/1pGgAdDeHEclh7nr_MX_RkryCgyf2IEtUQGyfbYMAh9k/edit#heading=h.oymnw3nlvwib

### CODE REVIEW

File: lib/elasticity/index_mapper.rb [hunk 1]
```diff
@@ -71,10 +72,16 @@ def remap!(retry_delete_on_recoverable_errors: true, retry_delay: 30, max_delay:
     end
 
     # Flushes the index, forcing any writes
+    # note that v7 no longer forces any writes on flush
     def flush_index
```
Comment (rstawarz) [inline lib/elasticity/index_mapper.rb]: would emitting a deprecation log warning be useful here, or is `flush_index` still have value if it doesn't force the writes?

File: lib/elasticity/strategies/alias_index.rb [hunk 2]
```diff
@@ -279,11 +288,20 @@ def mapping
 
       private
 
+      def build_index_name
+        ts = String.new
+        if @use_new_timestamp_format == true
+          ts = Time.now.utc.strftime("%Y%m%d%H%M%S%6N")
```
Comment (rstawarz) [inline lib/elasticity/strategies/alias_index.rb]: Any chance we could add some delimiters to make this easier to parse by humans reading the index names?  Something like `2019_10_27_211410` (not sure if `_` is an acceptable delimiter, but whatever is could be useful here.

File: lib/elasticity/index_mapper.rb (hunk 1 above)
Comment (davidbl) [reply lib/elasticity/index_mapper.rb]: `flush` is still a valid `thing`
https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-flush.html#flush-api-desc

 it just want have the side-effect of refreshing the index which made new documents visible to searches (the side-effect that is has pre v7)

File: lib/elasticity/strategies/alias_index.rb (hunk 2 above)
Comment (davidbl) [reply lib/elasticity/strategies/alias_index.rb]: yeah, we could.   but this format should be familiar to rails devs
https://github.com/doximity/doximity/tree/master/db/migrate  (which is why I picked it)

Comment (davidbl) [reply lib/elasticity/strategies/alias_index.rb]: and I was a bit scared of some future upgrade declaring other chars `off limits`

Comment (rstawarz) [reply lib/elasticity/strategies/alias_index.rb]: Yeah - I was thinking that same thing about the chars too.  I'm good either way, just figured I'd mention it.

Comment (davidbl) [reply lib/elasticity/strategies/alias_index.rb]: 👍 
fwiw,  most `human interaction` is usually via the aliases anyway

**Comment by davidbl** [comment]
@divoxx   yes.  I made the v7-specific stuff opt-in so, by default,  things should be compatible with older ES versions.

But we probably need to consider sun-setting support for truly old versions

**Comment by davidbl** [comment]
The testing of these changes is complete (see https://github.com/doximity/doximity/pull/30797)

And I have removed the `pre` tag and pushed the gem 
Screen Shot 2019-11-08 at 7 11 30 AM

**Pull request merged**